src/cmd-buildupload

```diff
@@ -193,7 +191,7 @@ def s3_check_exists(bucket, key, dry_run=False, endpoint_url=None):
 
 
 @retry(stop=retry_stop, retry=retry_boto_exception, retry_error_callback=retry_callback)
-def s3_copy(src, bucket, key, max_age, acl, extra_args={}, dry_run=False, endpoint_url=None):
+def s3_copy(src, bucket, key, max_age, acl, s3_client, extra_args={}, dry_run=False):
```

Comment:
And here too.

Reply:
Another approach would be to make a tiny S3 class which we initialize with the endpoint and dry run and then it only has two methods (`copy` and `check_exists`). But cool with this too!

Reply:
I can move the client real quick. Having it like that would be a bit more intuitive.

Regarding an S3 class, what about consolidating all of the S3 calls into a class at `cosalib.s3`? That way each of the `cmd`s that interact with S3 aren't doing their own thing.

Reply:
Oh yeah, kinda forgot about `cosalib.s3`. But yeah it makes sense to dedupe this stuff more.